pkg/cmd/roachtest/operations/changefeeds/create.go

```diff
@@ -0,0 +1,166 @@
+	err := updatePayloadForJobs(ctx, conn, allCFJobs)
+	if err != nil {
+		return err
+	}
+	options := make([]string, 0)
+	// Define the sink where the changefeed output will be sent.
+	sink, sinkOptions, err := getSinkConfigs(ctx, allCFJobs)
+	if err != nil {
+		return err
+	}
+	options = append(options, sinkOptions...)
+	// Calculate whether the new changefeed should include an initial scan.
+	scanOption, err := calculateScanOption(allCFJobs)
+	if err != nil {
+		return err
+	}
+	options = append(options, scanOption)
+
+	o.Status(fmt.Sprintf("creating changefeed job to sink %s with options %v", sink, options))
+
+	// Construct and execute the SQL statement to create the changefeed.
+	_, err = conn.ExecContext(ctx, fmt.Sprintf("CREATE CHANGEFEED FOR TABLE %s.%s INTO '%s' WITH %s;",
+		dbName, tableName, sink, strings.Join(options, ",")))
+	return err
+}
+
+// getSinkConfigs returns the sink uri along with the options for creating the changefeed.
+// this will be extended later for more sinks
+func getSinkConfigs(_ context.Context, _ []*jobDetails) (string, []string, error) {
+	return "null://", make([]string, 0), nil
+}
+
+// calculateScanOption determines whether the new changefeed should have an initial scan based on existing jobs.
+func calculateScanOption(allCFJobs []*jobDetails) (string, error) {
+	scanOnCountPercent := 0
+	if len(allCFJobs) > 0 {
+		scanOnCount := 0
+		// Count the number of jobs that have initial scan set to 'yes' or "only".
+		for _, j := range allCFJobs {
+			if v, ok := j.payload.Opts[changefeedbase.OptInitialScan]; ok {
+				if v == "yes" || v == "only" {
+					scanOnCount++
+				}
+			}
+		}
+		// Calculate the percentage of jobs with initial scan 'yes' or 'only'.
+		scanOnCountPercent = scanOnCount * 100 / len(allCFJobs)
+	}
+	// Get the maximum percentage of changefeeds that can have initial scans enabled.
+	maxPctScanOn, err := helpers.EnvOrDefaultInt(maxPctChangeFeedsScanOn, defaultEnvValuesInt[maxPctChangeFeedsScanOn])
+	if err != nil {
+		return "", err // Return error if fetching environment variable fails.
+	}
+
+	// Randomly decide whether the new changefeed should have an initial scan based on the max percentage allowed.
+	r, _ := randutil.NewPseudoRand()
+	initialScanValue := "no"
+	if scanOnCountPercent < maxPctScanOn && randutil.RandIntInRange(r, 0, 100) < maxPctScanOn {
+		initialScanValue = "yes"
+		// this will select initial_scan as "yes" or "only" with 50% probability.
+		if randutil.RandIntInRange(r, 0, 100) < 50 {
+			initialScanValue = "only"
+		}
+	}
+	resolved := ""
+	if initialScanValue != "only" {
+		resolved = ",resolved"
+	}
+	return fmt.Sprintf("%s='%s'%s", changefeedbase.OptInitialScan, initialScanValue, resolved), nil
+}
+
+// waitForJobToCancel waits until the job's status becomes "canceled" or "failed".
+func waitForJobToCreate(ctx context.Context, conn *gosql.DB, createdJob *jobDetails) error {
```

Comment:
this appears unused

Reply:
Yes @asg0451 , I am yet to get the change done.  =I was waiting for clarity on https://github.com/cockroachdb/cockroach/pull/140582#discussion_r1948845661. Please let me know if this is the right approach. Thanks